storm-core/src/jvm/org/apache/storm/command/HealthCheck.java

```diff
@@ -0,0 +1,124 @@
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class HealthCheck {
+    private static final String FAILED = "failed";
+    private static final String SUCCESS = "success";
+    private static final String TIMEOUT = "timeout";
+    private static final String FAILED_WITH_EXIT_CODE = "failed_with_exit_code";
+
+    public static int healthCheck(Map conf) {
+        String healthDir = ConfigUtils.absoluteHealthCheckDir(conf);
+        List<String> results = new ArrayList<>();
+        if (healthDir != null) {
+            File parentFile = new File(healthDir);
+            List<String> healthScripts = new ArrayList<String>();
+            if (parentFile.exists()) {
+                File[] list = parentFile.listFiles();
+                for (File f : list) {
+                    if (!f.isDirectory() && f.canExecute())
+                        healthScripts.add(f.getAbsolutePath());
+                }
+            }
+            for (String script : healthScripts) {
+                String result = processScript(conf, script);
+                results.add(result);
+            }
+        }
+
+        // failed_with_exit_code is OK. We're mimicing Hadoop's health checks.
+        // We treat non-zero exit codes as indicators that the scripts failed
+        // to execute properly, not that the system is unhealthy, in which case
+        // we don't want to start killing things.
+
+        if (results.contains(FAILED) || results.contains(TIMEOUT)) {
+            return 1;
+        } else {
+            return 0;
+        }
+
+    }
+
+    public static String processScript(Map conf, String script) {
+        Thread interruptThread = null;
+        try {
+            Process process = Runtime.getRuntime().exec(script);
+            final long timeout = (long) (conf.get(Config.STORM_HEALTH_CHECK_TIMEOUT_MS));
+            final Thread curThread = Thread.currentThread();
+            // kill process when timeout
+            interruptThread = new Thread(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        Thread.sleep(timeout);
+                        curThread.interrupt();
+                    } catch (InterruptedException e) {
+
+                    }
+                }
+            });
+            interruptThread.start();
+            process.waitFor();
+            interruptThread.interrupt();
+
+            if (process.exitValue() != 0) {
+                String str;
+                InputStream stdin = process.getInputStream();
+                BufferedReader reader = new BufferedReader(new InputStreamReader(stdin));
+                while ((str = reader.readLine()) != null) {
+                    if (StringUtils.isBlank(str)) {
+                        continue;
+                    }
```

Comment:
I'm not sure this is needed.  It doesn't hurt things, but str should never be null because of the while, and if it is the empty string it will not start with "ERROR"